crates/core/src/factories/mem_transport/test.rs

```diff
@@ -0,0 +1,368 @@
+            String::from_utf8_lossy(msg),
+            self.track.lock().unwrap(),
+        )))
+    }
+}
+
+async fn gen_tx(hnd: DynTxHandler) -> Transport {
+    let builder = Arc::new(crate::default_builder());
+    let hnd = TxImpHnd::new(hnd);
+    builder
+        .transport
+        .create(builder.clone(), hnd)
+        .await
+        .unwrap()
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn transport_notify() {
+    let h1 = TrackHnd::new();
+    let t1 = gen_tx(h1.clone()).await;
+    t1.register_space_handler(S1.clone(), h1.clone());
+    let u1 = h1.url();
+
+    let h2 = TrackHnd::new();
+    let t2 = gen_tx(h2.clone()).await;
+    t2.register_space_handler(S1.clone(), h2.clone());
+    let u2 = h2.url();
+
+    t1.send_space_notify(
+        u2.clone(),
+        S1.clone(),
+        bytes::Bytes::from_static(b"hello"),
+    )
+    .await
+    .unwrap();
+
+    t2.send_space_notify(
+        u1.clone(),
+        S1.clone(),
+        bytes::Bytes::from_static(b"world"),
+    )
+    .await
+    .unwrap();
+
+    h1.check_connect(&u2).unwrap();
+    h2.check_connect(&u1).unwrap();
+    h1.check_notify(&u2, &S1, b"world").unwrap();
+    h2.check_notify(&u1, &S1, b"hello").unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn transport_module() {
+    let h1 = TrackHnd::new();
+    let t1 = gen_tx(h1.clone()).await;
+    t1.register_module_handler(S1.clone(), "test".into(), h1.clone());
+    let u1 = h1.url();
+
+    let h2 = TrackHnd::new();
+    let t2 = gen_tx(h2.clone()).await;
+    t2.register_module_handler(S1.clone(), "test".into(), h2.clone());
+    let u2 = h2.url();
+
+    t1.send_module(
+        u2.clone(),
+        S1.clone(),
+        "test".into(),
+        bytes::Bytes::from_static(b"hello"),
+    )
+    .await
+    .unwrap();
+
+    t2.send_module(
+        u1.clone(),
+        S1.clone(),
+        "test".into(),
+        bytes::Bytes::from_static(b"world"),
+    )
+    .await
+    .unwrap();
+
+    h1.check_connect(&u2).unwrap();
+    h2.check_connect(&u1).unwrap();
+    h1.check_mod(&u2, &S1, "test", b"world").unwrap();
+    h2.check_mod(&u1, &S1, "test", b"hello").unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn transport_disconnect() {
+    let h1 = TrackHnd::new();
+    let t1 = gen_tx(h1.clone()).await;
+    t1.register_space_handler(S1.clone(), h1.clone());
+    let u1 = h1.url();
+
+    let h2 = TrackHnd::new();
+    let t2 = gen_tx(h2.clone()).await;
+    t2.register_space_handler(S1.clone(), h2.clone());
+    let u2 = h2.url();
+
+    t1.send_space_notify(
+        u2.clone(),
+        S1.clone(),
+        bytes::Bytes::from_static(b"hello"),
+    )
+    .await
+    .unwrap();
+
+    h2.check_connect(&u1).unwrap();
+
+    t1.disconnect(u2.clone(), Some("test-reason".into())).await;
+
+    h1.check_disconnect(&u2, Some("test-reason")).unwrap();
+    h2.check_disconnect(&u1, Some("test-reason")).unwrap();
```

Comment:
What happens after a disconnect if you send another space_notify?

Reply:
It would reconnect.